raco/myrial/parser.py

```diff
@@ -611,11 +611,15 @@ def p_file_parser_fun(p):
         """file_parser_fun : CSV LPAREN \
    schema_fun COMMA option_list RPAREN
  | CSV LPAREN schema_fun RPAREN
- | OPP LPAREN RPAREN"""
+ | OPP LPAREN RPAREN
+ | TIPSY LPAREN option_list RPAREN
+ | TIPSY"""
         if len(p) == 7:
             format, schema, options = (p[1], p[3], dict(p[5]))
         elif len(p) == 5:
-            format, schema, options = (p[1], p[3], {})
+            format = p[1]
+            schema = p[3] if p[1] != 'TIPSY' else []
```

Comment:
Can we add a TODO comment to split this into separate productions for each format?